packages/protocol/scripts/build.ts

```diff
@@ -103,14 +105,31 @@ function compile(outdir: string) {
   for (const externalContractPackage of externalContractPackages) {
     console.log(`Building external contracts for ${externalContractPackage.name}`)
     exec(
-      `yarn run truffle compile --silent --contracts_directory=./lib/${externalContractPackage.path}/contracts --contracts_build_directory=./build/contracts-${externalContractPackage.name}`
+      `yarn run truffle compile --silent --contracts_directory=./lib/${externalContractPackage.path}/contracts --contracts_build_directory=./build/contracts-${externalContractPackage.name}` // todo change to outdir
     )
   }
 
+  // compile 0.8 contracts
```

Comment:
Since we're doing similar things here and for external libs, would it make sense to abstract this away to a helper function? And use it also for the regular build for consistency.

Reply:
yes, I'm considering this refactor as well. Haven't done yet as it needs a couple more generalization for it to work with 0.8.